packages/wallets/src/features/cfd/modals/DynamicLeverageModal/DynamicLeverageModal.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useDynamicLeverage } from '@deriv/api';
+import { ModalWrapper, Transition, WalletText } from '../../../../components';
+import { useModal } from '../../../../components/ModalProvider';
+import useDevice from '../../../../hooks/useDevice';
+import BackArrow from '../../../../public/images/ic-back-arrow.svg';
+import { DynamicLeverageMarketCard } from '../../screens/DynamicLeverage/DynamicLeverageMarketCard';
+import { JurisdictionModal } from '../JurisdictionModal';
+import './DynamicLeverageModal.scss';
+
+const DynamicLeverageModal = () => {
+    const { data: dynamicLeverages } = useDynamicLeverage();
+    const { isMobile } = useDevice();
+    const { setModalState, show } = useModal();
+
+    return (
+        <ModalWrapper hideCloseButton>
+            <div className='wallets-dynamic-leverage-modal'>
+                <div className='wallets-dynamic-leverage-modal__title'>
+                    <BackArrow
+                        className='wallets-dynamic-leverage-modal__title-back'
+                        data-testid='back_icon'
+                        onClick={() => {
```

Comment:
@arshad-rao-deriv would be better to create onClickHandler above the JSX, it's easier to read and maintain components. Feel free to ignore this comment, it's something about personal preferences

Reply:
I agree with @sergei-deriv. This should be a function declare outside of the JSX. There are already to much things going on here.

Reply:
Updated.